spring-ws-core/src/main/java/org/springframework/ws/transport/http/HttpUrlConnection.java

```diff
@@ -118,13 +118,12 @@ protected long getResponseContentLength() throws IOException {
 
 	@Override
 	public Iterator<String> getResponseHeaders(String name) throws IOException {
-		String headerField = connection.getHeaderField(name);
-		if (headerField == null) {
-			return Collections.<String>emptyList().iterator();
-		}
-		else {
-			Set<String> tokens = StringUtils.commaDelimitedListToSet(headerField);
-			return tokens.iterator();
+		Map<String, List<String>> mapHeader = connection.getHeaderFields();
```

Comment:
This part of the patch looks fine and can be merged, with the small correction mentioned below.